### PR TITLE
Require use strict globally

### PR DESCRIPTION
**Pull request by esoergel**
https://eslint.org/docs/latest/rules/strict

## Technical Summary
Configure eslint to tell us to "use strict" globally.  This is going to raise a warning on a lot of files.

### Safety story
Linter config file only - no effect on production code.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by jingcheng16** [comment]
Hi Ethan, do we need to run any command to override the existing rules locally? I'm happy to test do we see warnings.

**Comment by jingcheng16** [comment]
:disappointed: Something unrelated to this PR... I just find out [we will error when the naming doesn't follow camelCase](https://github.com/dimagi/commcare-hq/blob/525ebf413fa8d5813b2dd4e6b09da16e3b9bc00a/.eslintrc.js#L46), but in one of my recent PR, I didn't use camelCase and I didn't receive an error message. I'm wondering if this line of config works...

**Comment by esoergel** [comment]
Looks like we do receive warnings on PRs:
image
They show up in the diff view and in the build logs, but don't make the lint job :x: .
I'm going with that approach for now for ease of introduction - happy for us to change it later.

**Comment by esoergel** [comment]
> in one of my recent PR, I didn't use camelCase and I didn't receive an error message. I'm wondering if this line of config works...

@jingcheng16 I think it depends on the circumstances, like it might only complain if you define a variable in snake case, but not if you reference one.

File: .eslintrc.js [hunk 1]
```diff
@@ -65,6 +65,7 @@ module.exports = {
         "space-before-blocks": ["error"],
         "space-in-parens": ["error", "never"],
         "space-infix-ops": ["error"],   // match flake8 E225
+        "strict": ["warn", "global"],
```
[the remark this answers is not in the document]
Comment (esoergel) [reply .eslintrc.js:69]: update: we do, see my comment on the main thread

**Review by jingcheng16 — APPROVED** [review]
Looks good to me. Thank you for making this PR!!

**Review by millerdev — APPROVED** [review]

**Pull request merged**